storage/filesystem/object.go

```diff
@@ -36,9 +37,66 @@ func (s *ObjectStorage) Set(core.Object) (core.Hash, error) {
 	return core.ZeroHash, fmt.Errorf("not implemented yet")
 }
 
+func (s *ObjectStorage) Get(h core.Hash) (core.Object, error) {
+	var obj core.Object
+	var err error
+
+	obj, err = s.getFromObject(h)
+	if err == nil {
+		return obj, nil
+	}
+
+	obj, err = s.getFromPackfile(h)
+	if err == nil {
+		return obj, nil
+	}
+
+	return nil, err
+}
+
+// Get returns the object with the given hash, by searching for it in
+// the git object directories.
+func (s *ObjectStorage) getFromObject(h core.Hash) (core.Object, error) {
```

Comment:
You must name your return values here, otherwise the defer function cannot overwrite the error value.